### DISCUSSION TITLE
Full Node State Sync Order

### DISCUSSION THREAD

**Discussion by ChiHaoLu** [NethermindEth/juno#800] [Q&A]
We can observe that a Full Node can synchronize its state from three sources: the Sequencer, P2P (other Full Nodes), and Ethereum. If the contents of these three sources differ, how would a Full Node decide the current state?

Based on my understanding:

- When a Full Node synchronizes a new transaction pending from the Sequencer, it adds this new transaction to the Pending state.
- Once the Sequencer successfully executes the transaction and obtains the state_diff, the transaction enters the Accepted on Layer2 state. At this point, the new state will be updated in the Local DB of the running Node.
- After the Layer1 Core (Rollup) Contract updates the state_diff, the transaction enters the Accepted on Layer1 state.

I believe there might be inaccuracies in my description, and I'm unsure about the correct approach. I would appreciate it if you could correct my description. Thank you very much!

**Comment by IronGauntlets** [comment]
We responded to this question in one of our discord channel. Therefore, I am closing this discussion.
